### DISCUSSION TITLE
Toper Cafemino modbus error

### DISCUSSION THREAD

**Discussion by Cafemino** [artisan-roaster-scope/artisan#677] [General]
Hello,
I just started with Toper Cafemino roaster, but I am not able to connect it to Artisan. Modbus error. I am using macbook with 10.13 high sierra, artisan v. 2.4.2. Cafemino with usb out, no communication. Previous owner used cropster with no issues, but told me that he was not able to connect with artisan.
Thanks!!!!!

**Comment by poundy** [comment]
Do you know what the settings the previous owner used with Cropster? 

The Toper USB profile is pretty simple, see https://github.com/artisan-roaster-scope/artisan/blob/master/src/includes/Machines/Toper/USB.aset for the settings it adopts when you select that. Without knowing whether there is something different/special about your Cafemino there's always a challenge getting things to connect correctly when nobody knows what the settings on your machine actually are....

**Reply by Cafemino** [reply]
Thank you for your answer, it is a challenge 🙂 I am thinking to add a separate device with two K probes anyway, to get more accuracy and easier connectivity, as I see that man people are having issues with different toper machines to connect.

**Comment by CarefreeBuzzBuzz** [comment]
And did you even run the Machine Configuration?   Please upgrade to the latest version as well.  If you want detailed help you really need to be very specific about what steps you took and what you experienced.  Thanks.  

___________________________
Michael Herbert, CarefreeBuzzBuzz

**Reply by Cafemino** [reply]
Hi yes I did run the machine config, tried both usb and ptc. Unfortunately my laptop does not allow to run 2.4.6 version due to os version, which cannot be upgraded too due to hw level (2012 laptop). Tried also the network manual configuration, no difference. Also tried on another laptop with 2.4.6. Version of Artisan, same. Anyway thank you for your kind support!

**Comment by CarefreeBuzzBuzz** [comment]
Ask Toper for the settings then, and do a factory reset and enter them manually.  
----------------------------------------
Michael Herbert, Head Roaster
Order at www.carefreebuzzbuzz.com <http://www.carefreebuzzbuzz.com/>
Artisan RoasterScope  <https://artisan-scope.org/>Team

**Comment by MAKOMO** [comment]
There are various variants of the Toper Cafemino. The latest feature a Schneider Modicon PLC and connect via the network/ethernet. Earlier models came with USB connecters, but all require different MODBUS setups. Ask the previous owner for his MODBUS on Cropster setup might be the fastest path to success.

**Comment by Cafemino** [comment]
Sp I opened my Cafemino and there is Phidgets!🤓 Installed the driver to my laptop, but there is a strange error occuring when trying to check the probes just from the phidgets utility. Any idea what is wrong? Thanks! For info, my cafemino is 2018.
![A7945061-CBBC-4906-9235-AEFDA041E5C2](https://user-images.githubusercontent.com/89744144/131383693-ae80f0ed-3f6d-45b9-aeaf-cc889234cb10.jpeg)
![F370A58A-203E-48AA-BA50-04D262513C2D](https://user-images.githubusercontent.com/89744144/131383697-2246be5a-2b75-4522-a6bb-631df3edf964.jpeg)

**Comment by CarefreeBuzzBuzz** [comment]
You cannot run Artisan and the Phidgets control panel at the same time

Michael Herbert 
CarefreeBuzzBuzz

**Reply by Cafemino** [reply]
Thanks a lot! It is on now!!!! I am really happy as the previous owner was never able to run artisan🤣probably never opened the machine🤓anyway sorry for amateurism and thanks all for help! Now I will just learn roasting✌️
